### DISCUSSION TITLE
working with patches

### DISCUSSION THREAD

**Discussion by adrianschroeter** [openSUSE/openSUSE-git#4] [General]
You write

"For example, the "git-example-1"[5] from Adrian is storing in the git
repository only the information about the package, delegating into OBS
the download of the pristine tarball (via #!RemoteAsset from pbuild).
"
why is that not an option? you can easily download the assets (eg "osc da") and work with tools like quilt. No?

**Comment by aplanas** [comment]
Uhm how would be the user experience here? I mean, you are in git, not in a valid osc checkout.

The user would need to have two directories, one for the git repository and the other for the osc checkout, go to the osc one and do the quilt magic, copy the patches back into the git repo and commit them from there. IMHO would be a real pain.

**Comment by adrianschroeter** [comment]
On Dienstag, 14. Juni 2022, 15:53:31 CEST Alberto Planas wrote:
> Uhm how would be the user experience here? I mean, you are in git, not in a valid osc checkout.
> 
> The user would need to have two directories, one for the git repository and the other for the osc checkout, go to the osc one and do the quilt magic, copy the patches back into the git repo and commit them from there. IMHO would be a real pain.

IMHO this can be in the same directory, if we automatically maintain
the .gitignore file. What we may have to do anyway for .osc and friends.

-- 

Adrian Schroeter ***@***.***>
Build Infrastructure Project Manager

SUSE Software Solutions Germany GmbH, Frankenstraße 146, 90461 Nürnberg, Germany
(HRB 36809, AG Nürnberg) Geschäftsführer: Ivo Totev

**Comment by mcepl** [comment]
Well, I very strongly believe that packaging and upstream should be kept separate (and thus, for example, I really dislike when a project wants to maintain `debian/` subdirectory or `*.spec` file in the upstream repo; hi, OpenStack!), those are two separate activities, respectively, packaging metadata should contain only distro-specific properties, thus by definition it should be nonsensical to keep one set of such properties for multiple distro. From my side, therefore, -1 to somehow mixing the two together.

However, when these two are kept separate I have developed and I use two different workflows:

1. https://src.opensuse.org/mcepl_pkgs/vis-master … using `#!RemoteAssetUrl: git+https://git.cepl.eu/git/vis/vis#devel` I point to the `devel` branch of my personal fork https://git.cepl.eu/cgit/vis/vis. All remotes are periodically refreshed and the `devel` branch is rebased on the top of `origin/master` with whatever additional patches I want to test. Works fine for me, but I am afraid the constant rewriting of this branch would not make our ISO-9000-types happy.
2. https://build.opensuse.org/package/show/devel:languages:python:Factory/python36 … which looks exactly like normal OBS repo, and it in fact is (and it is then mirrored to all our SUSE distros), but patches are generated [by script](https://src.opensuse.org/mcepl_tools/gosc/src/branch/main/git-format-named-patches.sh) from [Git branch](https://github.com/openSUSE-Python/cpython/tree/opensuse_3.6) … the key distinguishing feature is that the patches are named according to `Patch:` trailer in each commit (so that they don’t look automatically generated, and it is possible to convert to this style existing package).
